frontend/pages/hosts/ManageHostsPage/components/NoHosts/NoHosts.tsx

```diff
@@ -9,41 +9,62 @@ import RoboDogImage from "../../../../../../assets/images/robo-dog-176x144@2x.pn
 interface INoHostsProps {
   toggleGenerateInstallerModal: () => void;
   canEnrollHosts?: boolean;
+  includesSoftwareOrPolicyFilter?: boolean;
 }
 
 const baseClass = "no-hosts";
 
 const NoHosts = ({
   toggleGenerateInstallerModal,
   canEnrollHosts,
+  includesSoftwareOrPolicyFilter,
 }: INoHostsProps): JSX.Element => {
+  const renderContent = () => {
+    if (includesSoftwareOrPolicyFilter) {
+      return (
+        <div>
+          <h1>No hosts match the current criteria</h1>
+          <p>
+            Expecting to see new hosts? Try again in a few seconds as the system
+            catches up.
+          </p>
+        </div>
+      );
+    } else if (canEnrollHosts) {
```

Comment:
A nit, but a new pattern we implemented while you were out. Since you are returning from the first `if` you don't need the `else if` here. Could be: 

```
if (condition) {
  return response1;
}

if (otherCondition) {
  return response2;
}

return defaultResponse;
```

In general `else` statements are never really needed because the function can be short-circuited with `return` instead and overall makes the code easier to read.

Reply:
Oh yeah that make sense. I'm not sure why I didn't catch that 🤦🏽‍♂️. Editing now.